modules/P02_firmware_bin_file_check.sh

```diff
@@ -107,62 +111,76 @@ fw_bin_detector() {
   if [[ "$AVM_CHECK" -gt 0 ]] || [[ "$FW_VENDOR" == *"AVM"* ]]; then
     print_output "[*] Identified AVM firmware - using AVM extraction module"
     export AVM_DETECTED=1
+    write_csv_log "AVM firmware detected" "yes" "NA"
   fi
   # if we have a zip, tgz, tar archive we are going to use the patools extractor
   if [[ "$FILE_BIN_OUT" == *"gzip compressed data"* || "$FILE_BIN_OUT" == *"Zip archive data"* || "$FILE_BIN_OUT" == *"POSIX tar archive"* || "$FILE_BIN_OUT" == *"ISO 9660 CD-ROM filesystem data"* ]]; then
     # as the AVM images are also zip files we need to bypass it here:
     if [[ "$AVM_DETECTED" -ne 1 ]]; then
       print_output "[*] Identified gzip/zip/tar/iso archive file - using patools extraction module"
       export PATOOLS_INIT=1
+      write_csv_log "basic compressed (patool)" "yes" "NA"
     fi
   fi
   if [[ "$FILE_BIN_OUT" == *"VMware4 disk image"* ]]; then
     print_output "[*] Identified VMWware VMDK archive file - using VMDK extraction module"
     export VMDK_DETECTED=1
+    write_csv_log "VMDK" "yes" "NA"
   fi
   if [[ "$FILE_BIN_OUT" == *"UBI image"* ]]; then
     print_output "[*] Identified UBI filesystem image - using UBI extraction module"
     export UBI_IMAGE=1
+    write_csv_log "UBI filesystem" "yes" "NA"
   fi
   if [[ "$DLINK_ENC_CHECK" == *"SHRS"* ]]; then
     print_output "[*] Identified D-Link SHRS encrpyted firmware - using D-Link extraction module"
     export DLINK_ENC_DETECTED=1
+    write_csv_log "D-Link SHRS" "yes" "NA"
   fi
   if [[ "$DLINK_ENC_CHECK" =~ 00000000\ \ 00\ 00\ 00\ 00\ 00\ 00\ 0.\ ..\ \ 00\ 00\ 0.\ ..\ 31\ 32\ 33\ 00 ]]; then
-    print_output "[*] Identified Engenius encrpyted firmware - using Engenius extraction module"
+    print_output "[*] Identified EnGenius encrpyted firmware - using Engenius extraction module"
```

Comment:
If you fix the first `EnGenius` occurrence in this string, you also should fix the second one 😆